databuilder/query_engines/sqlite.py

```diff
@@ -151,9 +282,101 @@ def engine(self):
         return engine
 
 
-def get_tables(obj):
-    if isinstance(obj, sqlalchemy.sql.ClauseElement):
-        return get_referenced_tables(obj)
+def prepare_query(query):
+    query = define_placeholder_tables(query)
+    query = apply_patient_joins(query)
+    return query
+
+
+def define_placeholder_tables(query):
```

Comment:
Could this be called "replace_..." or "insert_..." or something to make it clear that it's returning a modified query?